src/blog/dark-mode-with-style-dictionary/index.mdx

```diff
@@ -0,0 +1,1352 @@
+
+<ThemedImage style={{width: '80%', marginLeft: '10%'}}>
+
+![Xcode showing asset catalog](images/ios-asset-light.png)
+![Xcode showing asset catalog](images/ios-asset-dark.png)
+
+</ThemedImage>
+
+If you take a look at the file structure, that asset catalog looks like this:
+
+<FileTree items={[{
+  name: 'Assets.xcassets',
+  type: 'folder',
+  items: [{
+    name: 'backgroundPrimary.colorset',
+    type: 'folder',
+    items: [{
+      type: 'json',
+      name: 'Contents.json'
+    }]
+  },{
+    name: 'logo.imageset',
+    type: 'folder',
+    items: [{
+      type: 'json',
+      name: 'Contents.json'
+    },{
+      type: 'image',
+      name: 'logo.png'
+    },{
+      type: 'image',
+      name: 'logo-dark.png'
+    }]
+  },{
+    name: 'Contents.json',
+    type: 'json'
+  }]
+}]} />
+
+If you want to dive deeper on asset catalogs I suggest this article: [Working with Assets Catalogs in Xcode](https://learnappmaking.com/xcode-assets-catalogs-how-to/).
+
+### Colors
+
+A colorset contains a single **Contents.json** file which describes the color and potentially different values for the color in different contexts like light and dark mode. Here is a color with light and dark mode:
+
+```json
+{
+  "colors": [
+    {
+      "idiom": "universal",
+      "color": {
+        "color-space": "srgb",
+        "components": {
+          "alpha": "1.0000",
+          "blue": "1.0000",
+          "red": "1.0000",
+          "green": "1.0000"
+        }
+      }
+    },
+    {
+      "idiom": "universal",
+      "color": {
+        "color-space": "srgb",
+        "components": {
+          "alpha": "1.0000",
+          "blue": "0.1255",
+          "red": "0.0863",
+          "green": "0.1255"
+        }
+      },
+      "appearances": [
+        {
+          "appearance": "luminosity",
+          "value": "dark"
+        }
+      ]
+    }
+  ],
+  "info": {
+    "author": "xcode",
+    "version": 1
+  }
+}
+```
+
+A few things to notice:
+
+- `colors` is an array of color objects because one color might have multiple values, one for each different context like light and dark mode.
+- A `color` can have an `appearances` array, which will let iOS know when that color should be used. There are 2 types of appearances you can have: luminosity and contrast. Luminosity can be 'light', 'dark', or 'any'. Contrast can be 'high' or nothing, meaning normal contrast. *(sidenote: this is kind of weird, from a user's perspective both contrast and luminosity are binary settings: light/dark and on/off for contrast. Why then are luminosity and contrast different for a developer?)*
```

Comment:
kind of weird, from a user's perspective both > kind of weird. From a user's perspective, both